flytekit/annotated/context_manager.py

```diff
@@ -78,6 +83,7 @@ def __init__(self, parent=None,
                  execution_state: ExecutionState = None,
                  flyte_client: flyte_client.SynchronousFlyteClient = None,
                  user_space_params: ExecutionParameters = None,
+                 branch_eval_mode: BranchEvalMode = None,
```

Comment:
What do you think about putting this into ExecutionState?

Reply:
makes sense will do